### ISSUE TITLE
Multiline string rejects line terminator elision

### ISSUE THREAD

**Issue by infogulch** [cue-lang/cue#3469]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.10.0

go version go1.23.0
      -buildmode exe
       -compiler gc
       -trimpath true
     CGO_ENABLED 0
          GOARCH amd64
            GOOS linux
         GOAMD64 v1
cue.lang.version v0.10.0
</pre>

### Does this issue reproduce with the latest stable release?

Yes

### What did you do?

Tried to evaluate the string from the spec under [String and byte sequence literals](https://cuelang.org/docs/reference/spec/#string-and-byte-sequence-literals):

```cue
"""
    lily:
    out of the water
    out of itself

    bass
    picking \
    bugs
    off the moon
        — Nick Virgilio, Selected Haiku, 1988
    """
```

### What did you expect to see?

According to the spec page it should be equivalent to:

```cue
"lily:\nout of the water\nout of itself\n\n" +
"bass\npicking bugs\noff the moon\n" +
"    — Nick Virgilio, Selected Haiku, 1988"
```

### What did you see instead?

<pre>
cannot get syntax for "-": unknown escape sequence:
    -:7:14
</pre>

### Notes

Here's a playground with this issue: https://cuelang.org/play/?id=7uimnxuUVGR#w=function&i=cue&f=eval&o=cue

A smaller reproduction case: https://cuelang.org/play/?id=hxhDZo_g4zz#w=function&i=cue&f=eval&o=cue

**Comment by infogulch** [comment]
Ah this seems related to https://github.com/cue-lang/cue/issues/2317

**Comment by mvdan** [comment]
Yes; this is #2317. Line continuations with backslashes are not implemented right now.
